images/router/haproxy/conf/haproxy-config.template

```diff
@@ -34,7 +34,7 @@
 
 {{- /* hsts header in response: */}}
 {{- $hstsOptionalTokenPattern := `(?:includeSubDomains|preload)` }}
-{{- $hstsPattern := printf `(?:%[1]s[;])*max-age=(?:\d+|"\d+")(?:[;]%[1]s)*`  $hstsOptionalTokenPattern -}}
+{{- $hstsPattern := printf `(?i)(?:%[1]s\s*[;]\s*)*max-age\s*=\s*(?:\d+|"\d+")(?:\s*[;]\s*%[1]s)*`  $hstsOptionalTokenPattern -}}
```

Comment:
Actually, as I read your unit tests, I realized this gets a little complicated.  In https://datatracker.ietf.org/doc/html/rfc6797#section-6.1 it says that it's okay to supply invalid directives, they just get ignored.  max-age is the only one that MUST be there.  So any other wrong directive is okay, even a wrongly spelled includeSubdomains or preload.  If max-age is missing (or spelled wrong), that would make it not match and no HSTS header printed.

Reply:
Per our conversation, we will probably be removing non-matching directives from the header in the API when we implement https://issues.redhat.com/browse/NE-310. I think the pattern should be okay as is, and if there are further issues with writing HSTS before NE-310, we have to remember this conversation.

Reply:
As discussed, I keep it the was it is adding the comment about the not full conformance